pennylane_rigetti/qc.py

```diff
@@ -94,29 +93,30 @@ def __init__(self, device, *, shots=1000, wires=None, active_reset=False, **kwar
 
         self.qc = self.get_qc(device, **timeout_args)
 
-        self.num_wires = len(self.qc.qubits())
-
-        if wires is None:
-            # infer the number of modes from the device specs
-            # and use consecutive integer wire labels
-            wires = range(self.num_wires)
+        qubits = sorted(
+            [
+                qubit.id
+                for qubit in self.qc.quantum_processor.to_compiler_isa().qubits.values()
+                if qubit.dead is False
+            ]
+        )
 
+        # Interpret wires as Pennylane would, and use the labels to map to actual qubits on the device.
         if isinstance(wires, int):
-            raise ValueError(
-                f"Device has a fixed number of {self.num_wires} qubits. The wires argument can only be used "
-                "to specify an iterable of wire labels."
-            )
-
-        if self.num_wires != len(wires):
-            raise ValueError(
-                f"Device has a fixed number of {self.num_wires} qubits and "
-                f"cannot be created with {len(wires)} wires."
-            )
-
-        self.wiring = dict(enumerate(self.qc.qubits()))
+            pennylane_wires = Wires(list(range(wires)))
+        elif isinstance(wires, Iterable):
+            pennylane_wires = Wires(wires)
+        else:
+            raise ValueError("wires must be an integer or an iterable of numbers or strings.")
+
+        self.wiring = {
```

Comment:
I'm not convinced that we need the `wiring` attribute anymore, unless it's used somewhere else I'm not aware of. and I think the more wire-maps we have, the more confusing this becomes. 

If we define a `self._qubits = qubits` so we have those accessible for later, and `num_wires` as `len(qubits)`, I think we can get rid of this entirely, assuming I'm correct about my suggested change to `define_wire_map` below.